packages/sitecore-jss-react/src/enhancers/withEmptyFieldEditingComponent.test.tsx

```diff
@@ -0,0 +1,196 @@
+      },
+      emptyFieldEditingComponent: EmptyFieldEditingComponent,
+    };
+
+    const WrappedComponent = withEmptyFieldEditingComponent<TestComponentProps>(TestComponent, {
+      defaultEmptyFieldEditingComponent: DefaultEmptyFieldEditingComponentText,
+    });
+
+    const rendered = mount(<WrappedComponent {...props} />);
+    const expected = mount(<EmptyFieldEditingComponent />);
+
+    expect(rendered.html()).to.equal(expected.html());
+  });
+
+  it('Should render component if field value is provided', () => {
+    const props = {
+      field: {
+        value: 'field value',
+        metadata: testMetadata,
+      },
+    };
+
+    const WrappedComponent = withEmptyFieldEditingComponent<TestComponentProps>(TestComponent, {
+      defaultEmptyFieldEditingComponent: DefaultEmptyFieldEditingComponentText,
+    });
+
+    const rendered = mount(<WrappedComponent {...props} />);
+    expect(rendered.html()).to.equal('<div><h1>field value</h1><h2>foo</h2><p>bar</p></div>');
+  });
+
+  it('Should render component if component is explicitly not editable if value is empty in metadata edit mode', () => {
+    const props = {
+      field: {
+        value: '',
+        metadata: testMetadata,
+      },
+      editable: false,
+    };
+
+    const WrappedComponent = withEmptyFieldEditingComponent<TestComponentProps>(TestComponent, {
+      defaultEmptyFieldEditingComponent: DefaultEmptyFieldEditingComponentText,
+    });
+
+    const rendered = mount(<WrappedComponent {...props} />);
+    expect(rendered.html()).to.equal('<div><h1></h1><h2>foo</h2><p>bar</p></div>');
+  });
+
+  it('Should render component if metadata is not provided', () => {
+    const props = {
+      field: {
+        value: '',
+      },
+    };
+
+    const WrappedComponent = withEmptyFieldEditingComponent<TestComponentProps>(TestComponent, {
+      defaultEmptyFieldEditingComponent: DefaultEmptyFieldEditingComponentText,
+    });
+
+    const rendered = mount(<WrappedComponent {...props} />);
+    expect(rendered.html()).to.equal('<div><h1></h1><h2>foo</h2><p>bar</p></div>');
+  });
+
+  it('Should render component with forward ref if field value is provided in metadata edit mode', () => {
+    const props = {
+      field: {
+        value: 'field value',
+        metadata: testMetadata,
+      },
+    };
+
+    const WrappedComponent = withEmptyFieldEditingComponent<TestComponentProps>(
+      TestComponentWithRef,
+      {
+        defaultEmptyFieldEditingComponent: DefaultEmptyFieldEditingComponentText,
+        isForwardRef: true,
+      }
+    );
+    const ref = React.createRef<HTMLDivElement>();
+    const rendered = mount(<WrappedComponent {...props} ref={ref} />);
+
+    expect(ref.current?.outerHTML).to.equal('<h2>foo</h2>');
+    expect(rendered.html()).to.equal('<div><h1>field value</h1><h2>foo</h2><p>bar</p></div>');
+  });
+
+  it('Should render component if field src value is provided (case for Image component)', () => {
```

Comment:
You need to add unit tests to cover the cases when _field_ it's a value itself. For Image, Link components. So I would recommend to add separate _describe_ blocks for these components